### PR TITLE
Feat/bump flutter svg

### LINKED ISSUE #272: [FEAT] [Flutter] update flutter_svg version to ^2.0.4
**Is your feature request related to a problem? Please describe.**
I got this error running `flutter pub get` due outdated `flutter_svg` version
```
Because every version of iconoir_flutter depends on flutter_svg ^1.1.1 and design depends on flutter_svg ^2.0.4, iconoir_flutter is forbidden.
So, because design depends on iconoir_flutter ^6.4.0, version solving failed.
```

**Describe the solution you'd like**
`flutter_svg: ^2.0.4` in `pubspec.yaml`

### PR DESCRIPTION
**Pull request by Ciock**
fixes #272 

`colorFilter: ColorFilter.mode(Colors.red, BlendMode.srcIn)` is the new way of defining a color in `flutter_svg` after `2.0.0` release.

---

**There was a naming ambiguity between the icon `ColorFilter` and `flutter/widgets.ColorFilter`**.

I resolved this way with alias `import 'package:flutter/widgets.dart' as w`, but I don't know if it could cause any problems in code autogeneration in future updates. Changing icon name would prevent it for sure.
```
import 'package:flutter/widgets.dart' as w;
import 'package:flutter_svg/flutter_svg.dart';

class ColorFilter extends w.StatelessWidget {
  final w.Color? color;
  final double? width;
  final double? height;

  const ColorFilter({w.Key? key, this.color, this.width, this.height})
      : super(key: key);

  @override
  w.Widget build(w.BuildContext context) => SvgPicture.string(
        '''
<svg width="24" height="24" stroke-width="1.5" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12 14.5C15.3137 14.5 18 11.8137 18 8.5C18 5.18629 15.3137 2.5 12 2.5C8.68629 2.5 6 5.18629 6 8.5C6 11.8137 8.68629 14.5 12 14.5Z" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M16 21.5C19.3137 21.5 22 18.8137 22 15.5C22 12.1863 19.3137 9.5 16 9.5C12.6863 9.5 10 12.1863 10 15.5C10 18.8137 12.6863 21.5 16 21.5Z" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M8 21.5C11.3137 21.5 14 18.8137 14 15.5C14 12.1863 11.3137 9.5 8 9.5C4.68629 9.5 2 12.1863 2 15.5C2 18.8137 4.68629 21.5 8 21.5Z" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
''',
        colorFilter: color != null
            ? w.ColorFilter.mode(color!, w.BlendMode.srcIn)
            : null,
        width: width,
        height: height,
      );
}

```

### CODE REVIEW

**Comment by Ciock** [comment]
I made these changes:

https://github.com/iconoir-icons/iconoir/blob/64b3cc2e6031fdc2b7b07ab09b192597145f244e/constants.js#L14-L17

In the `Building Flutter libraries` task only
https://github.com/iconoir-icons/iconoir/blob/64b3cc2e6031fdc2b7b07ab09b192597145f244e/bin/build.js#L339-L342

However, the color-filter icon in react library is sometimes renamed on the script run. 🧐
I don't know why it happens, it seems that the Flutter task overrides something that the React task reads. Do you know what it could be?

**Comment by sammarks** [comment]
I would assume it's because of the `dstFilePaths` being updated in the context (which is shared across all tasks). The build script could certainly be improved 😅

I would try renaming `dstFilePaths` to something different like `flutterDstFilePaths`, and then update the flutter-specific tasks to use that name, and see if that helps.

**Comment by Ciock** [comment]
I found it! It was due to React and Flutter tasks using the same temp directory 👍 

However, looking at `ctx` helped in finding the bug.

File: constants.js [hunk 1]
```diff
@@ -10,3 +10,8 @@ export const incompatibleNames = {
   tiktok: 'tikTok',
   youtube: 'youTube',
 };
+
+export const flutterIncompatibleNames = {
+  ...incompatibleNames,
+  'color-filter': 'color-filter-ic'
```
Comment (sammarks) [inline constants.js:16]:
```suggestion
  'color-filter': 'color-filter-icon'
```

Sorry for the nitpick, but probably best to establish adding "Icon" to the end of incompatible names. And then this should be good to go!

**Review by sammarks — CHANGES_REQUESTED** [review]

File: constants.js (hunk 1 above)
Comment (Ciock) [reply constants.js]: It's ok, I was unsure of what to call it 😅

**Review by sammarks — APPROVED** [review]
Looks good, thanks for the quick PR!

**Pull request merged**